diff --git a/src/ActionLink/ActionLinkTypeBase.php b/src/ActionLink/ActionLinkTypeBase.php
index e785ef5..42e52d0 100644
--- a/src/ActionLink/ActionLinkTypeBase.php
+++ b/src/ActionLink/ActionLinkTypeBase.php
@@ -13,6 +13,8 @@ use Drupal\Core\Session\AccountInterface;
 use Drupal\Core\StringTranslation\StringTranslationTrait;
 use Drupal\flag\FlagInterface;
 use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpFoundation\Request;
+
 
 /**
  * Provides a base class for all link types.
@@ -152,13 +154,25 @@ abstract class ActionLinkTypeBase extends PluginBase implements ActionLinkTypePl
    */
   protected function getDestination() {
     $current_url = Url::fromRoute('<current>');
+    $current_path = $current_url->getInternalPath();
     $route_params = $current_url->getRouteParameters();
 
     if (isset($route_params['destination'])) {
       return $route_params['destination'];
     }
 
-    return $current_url->getInternalPath();
+    // Attempt to get the parent path if this link is
+    // being built during a views AJAX request.
+    $current_query_params = \Drupal::request()->query->all();
+    if (!empty($current_query_params['ajax_page_state']) && count($current_query_params['ajax_page_state'])) {
+      // If the referrer is an internal path, use that instead.
+      $referrer_request = Request::create(\Drupal::request()->server->get('HTTP_REFERER'));
+      if ($referrer = \Drupal::service('path.validator')->getUrlIfValid($referrer_request->getRequestUri())) {
+        return $referrer->getInternalPath();
+      }
+    }
+
+    return $current_path;
   }
 
   /**
